src/main/java/kr/syeyoung/dungeonsguide/Main.java

```diff
@@ -33,110 +39,229 @@
-                    ProgressManager.pop(progressBar);
-                    isLoaded = true;
-                } catch (Throwable e) {
-                    cause = e;
-                    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-                    PrintStream printStream = new PrintStream(byteArrayOutputStream);
-                    e.printStackTrace(printStream);
-                    stacktrace = new String(byteArrayOutputStream.toByteArray());
-
-                    while (progressBar.getStep() < progressBar.getSteps())
-                        progressBar.step("random-"+progressBar.getStep());
-                    ProgressManager.pop(progressBar);
-
-                    e.printStackTrace();
+            try (InputStream premiumControlClass = this.getClass().getResourceAsStream("/kr/syeyoung/dungeonsguide/e.class")) {
+                progressBar = ProgressManager.push("DungeonsGuide", premiumControlClass == null ? 7 : 6);
+            }
+
+
+
+            AuthManager.getInstance().setBaseserverurl(SERVER_URL);
+
+
+            String version = null;
+            try (InputStream resourceAsStream = this.getClass().getResourceAsStream("/kr/syeyoung/dungeonsguide/DungeonsGuide.class")) {
+                if (resourceAsStream == null) {
+                    if (System.getProperty("dg.version") == null) {
+                        version = "nlatest";
+                    } else {
+                        version = System.getProperty("dg.version");
+                    }
                 }
             }
-        } catch (IOException  | AuthenticationException | NoSuchAlgorithmException | CertificateException | KeyStoreException | KeyManagementException | InvalidKeySpecException | SignatureException e) {
-            cause = e;
-            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-            PrintStream printStream = new PrintStream(byteArrayOutputStream);
-            e.printStackTrace(printStream);
-            stacktrace = new String(byteArrayOutputStream.toByteArray());
 
-            while (progressBar.getStep() < progressBar.getSteps())
-                progressBar.step("random-"+progressBar.getStep());
-            ProgressManager.pop(progressBar);
 
-            e.printStackTrace();
+            AuthManager.getInstance().init();
+
+
+            ResourceManager.getInstance().setBaseUrl(SERVER_URL);
+            ResourceManager.getInstance().setBASE64_X509ENCODEDKEYSPEC(SOME_FUNNY_KEY_THING);
+
+            if(!AuthManager.getInstance().isPlebUser()){
+                downloadAssets(version);
+            }
+
+            URL.setURLStreamHandlerFactory(new DGStreamHandlerFactory());
+            LaunchClassLoader classLoader = (LaunchClassLoader) Main.class.getClassLoader();
+            classLoader.addURL(new URL("z:///"));
+
+            progressBar.step("Initializing");
+            DungeonsGuide dungeonsGuide = DungeonsGuide.getDungeonsGuide();
```

Comment:
Move it to DungeonsGuide.java

We need to keep Main VERY general

We are not going to change Main often, because that causes autoupdate jar update

This Main is too specific. Delegate this to actual mod implementation DungeonsGuide.java